src/stream.iced

```diff
@@ -0,0 +1,114 @@
+stream = require('keybase-chunk-stream')
+nacl = require('keybase-nacl')
+armor = require('node-armor-x')
+msgpack = require('msgpack-lite')
+payload = require('./payload')
+header = require('./header')
+nonce = require('./nonce')
+format = require('./format')
+
+saltpack_block_len = (1024**2)
+noop = () ->
+
+class NaClEncryptStream extends stream.ChunkStream
+  _encrypt : (chunk) =>
+    payload_list = payload.generate_encryption_payload_packet(@_encryptor, chunk, @_block_num, @_header_hash, @_mac_keys)
+    ++@_block_num
+    return payload_list
+
+  _transform : (chunk, encoding, cb) ->
+    if not @_header_written
+      {header_intermediate, header_hash, mac_keys, payload_key} = header.generate_encryption_header_packet(@_encryptor, @_recipients, {anonymized_recipients : @_anonymized_recipients})
+      @_header_hash = header_hash
+      @_mac_keys = mac_keys
+      @_encryptor.secretKey = payload_key
+      @push(header_intermediate)
+      @_header_written = true
+    super(chunk, encoding, cb)
```

Comment:
Are you talking about line 14? no - this is the specification of the `@_encrypt` method. the `node-chunk-stream` contract is such that when you create a new `chunk-stream`, you specify one function that accepts a buffer and returns a buffer or an object (depending on if the stream is in object mode or not), which dictates the transformation that's done inside the `chunk-stream`. so here we're creating a new function that accepts plaintext and encrypts with the secret key given in the constructor for the `NaClEncryptStream`, and passing said function as the transform function.